components/webac/src/main/java/org/trellisldp/webac/WebAcFilter.java

```diff
@@ -163,6 +148,16 @@ public WebAcFilter(final WebAcService accessService, final List<String> challeng
                 stream(a.split(",")).map(String::trim).map(String::toUpperCase).forEach(appendable::add));
     }
 
+    /**
+     * Set the access service.
+     *
+     * @param accessService the access service
+     */
+    @Inject
+    public void setAccessService(final WebAcService accessService) {
```

Comment:
If someone wants to use these classes in a non CDI env changing this to a setter makes this class harder to use. There will be a NPE in the filter if they forget to set this dependency. Do you see any issue here? Should there be a default or Noop implementation of this dependency set in the constructor?

Reply:
This is exactly why I prefer ctor injection. The issue, however, is that the JAX-RS specification (independent of CDI) does not require support for ctor injection. So while Jersey and CXF both support ctor injection, RESTeasy does not, which actually makes these classes rather difficult to use with Thorntail, Quarkus or anything else built on RESTeasy.

So, in order to support the widest range of JAX-RS implementations, we will need either field or setter injection for these classes, and I find setter injection to be much easier to deal with than field injection.

That said, I don't have an opinion about adding another constructor that also sets this field. I would be a little concerned that it would be a bit confusing: do you set the object in the ctor or don't you? Certainly the documentation could clarify what is going on.

Is it reasonable to expect that someone using JAX-RS will be familiar with how injection works?